Remove Dashing, Eloquent, Foxy, L500 and SR300 support

SamerKhshiboun opened the pull request:
Remove EOL models
Remove EOL distros
Remove unused files/code (specially related to fisheye and pose sensors)

SamerKhshiboun committed:
support color and depth/ir formats

SamerKhshiboun committed:
remove unsupported models

SamerKhshiboun committed:
Merge branch 'ros2-development' into remove_l500_sr300

Nir-Az commented:
This is draft right?
Please update when it's ready for review.

SamerKhshiboun committed:
merge

SamerKhshiboun committed:
Merge branch 'IntelRealSense:ros2-development' into remove_l500_sr300

Arun-Prasad-V commented:
Hi @SamerKhshiboun ,

Since POSE stream is getting removed, does it make sense to remove the support for POSE sensor as well? 
Reference line:
https://github.com/IntelRealSense/realsense-ros/blob/19bc683074f8d757fec8925c15aea7317d401728/realsense2_camera/src/rs_node_setup.cpp#L156

SamerKhshiboun commented:
> sensor.is<rs2::pose_sensor>()

Fixed. Also deleted PoseProfile usage in other places

Arun-Prasad-V commented:
> > sensor.isrs2::pose_sensor()
> 
> Fixed. Also deleted PoseProfile usage in other places

@SamerKhshiboun, It's still present in profile_manager.cpp
https://github.com/IntelRealSense/realsense-ros/blob/844210d353ab8665bd245511edd9de19dbc383bf/realsense2_camera/src/profile_manager.cpp#L596

SamerKhshiboun commented:
> > > sensor.isrs2::pose_sensor()
> > 
> > 
> > Fixed. Also deleted PoseProfile usage in other places
> 
> @SamerKhshiboun, It's still present in profile_manager.cpp
> 
> https://github.com/IntelRealSense/realsense-ros/blob/844210d353ab8665bd245511edd9de19dbc383bf/realsense2_camera/src/profile_manager.cpp#L596

Fixed, somehow it was missed in the git add command

Nir-Az commented on realsense2_camera/scripts/rs2_listener.py:
```diff
@@ -19,18 +19,12 @@
-# import sensor_msgs.point_cloud2 as pc2
+import tf2_ros
+from sensor_msgs.msg import PointCloud2 as msg_PointCloud2
+from sensor_msgs_py import point_cloud2 as pc2
```
This used to be `humble` only, is it relevant to `iron` and `rolling` too?

SamerKhshiboun replied:
did not understand the question.. can you elaborate more ? are you talking about pc2 or tf2_ros ?

Nir-Az replied:
NVM I understand now :)

Nir-Az commented:
@SamerKhshiboun please solve conflicts `This branch has conflicts that must be resolved`

Arun-Prasad-V commented:
@SamerKhshiboun , Shall we remove the support of 'fisheye' sensor as well?
https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_camera/src/rs_node_setup.cpp#L146
https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_camera/src/rs_node_setup.cpp#L332

Also, in rs_launch.py
https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_camera/launch/rs_launch.py#L47

https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_description/urdf/_r430.urdf.xacro#L35

SamerKhshiboun commented:
> @SamerKhshiboun , Shall we remove the support of 'fisheye' sensor as well?
> 
> https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_camera/src/rs_node_setup.cpp#L146
> 
> 
> https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_camera/src/rs_node_setup.cpp#L332
> 
> Also, in rs_launch.py
> 
> https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_camera/launch/rs_launch.py#L47
> 
> https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_description/urdf/_r430.urdf.xacro#L35

Fixed

SamerKhshiboun committed:
rebase from ros2-development

SamerKhshiboun committed:
rebase from ros2-development

Nir-Az reviewed (approved):
LGTM

christian-rauch commented:
Is there a good reason to break the compatibility with `foxy`? Couldn't this be handled with #ifdefs like the other distributions? Could we alternatively at least introduce branches for distributions that are incompatible with the development branch? E.g. create a `foxy` branch for that last commit that still compiles on ROS `foxy`?
